include/rdma/fabric.h

```diff
@@ -168,6 +168,7 @@ typedef struct fid *fid_t;
 #define FI_PEER_TRANSFER	(1ULL << 36)
 #define FI_MR_DMABUF		(1ULL << 40)
 #define FI_AV_USER_ID		(1ULL << 41)
+#define FI_DIRECTED_AUTH_KEY	(1ULL << 42)
```

Comment:
One of the problems this capability is trying to solve is if FI_DIRECTEC_RECV can be support with multiple authorization keys.

Reply:
Can't that be determined by checking the auth_key_cnt (or max or whatever we called it) and whether FI_DIRECTED_RECV is enabled?  We require support for any-source.  Couldn't we extend that requirement to auth_keys?

Reply:
I think it can be determined based on max_ep_auth_key. The issue here is how can the provider advertise two different capability sets based on max_ep_auth_key? Maybe reporting multiple fi_info's is the solution here.

fi_info 1:
- max_ep_auth_key == 1
- caps |= FI_DIRECTED_RECV

fi_info 2:
- max_ep_auth_key > 2
- caps &= ~FI_DIRECTED_RECV

This would put the burden on the app to sort through if multiple fi_infos are reported for the same NIC. Thoughts?

Reply:
I'll take the approach of having to fi_info's based on max_ep_auth_key.